NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java

```diff
@@ -599,47 +607,9 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onPrepareOptionsMenu(Menu menu) {
         super.onPrepareOptionsMenu(menu);
-        final boolean haveItems = getListAdapter() != null && getListAdapter().getCount() > 0;
-
-        // If there are any notes in the list (which implies that one of
-        // them is selected), then we need to generate the actions that
-        // can be performed on the current selection. This will be a combination
-        // of our own specific actions along with any extensions that can be
-        // found.
-        if (haveItems) {
-            // This is the selected item.
-            Uri uri = ContentUris.withAppendedId(
-                    getIntent().getData(),
-                    getSelectedItemId()
-            );
 
-            // Build menu... always starts with the EDIT action...
-            Intent[] specifics = new Intent[1];
-            specifics[0] = new Intent(Intent.ACTION_EDIT, uri);
-            MenuItem[] items = new MenuItem[1];
-
-            // ... is followed by whatever other actions are available...
-            Intent intent = new Intent(null, uri);
-            intent.addCategory(Intent.CATEGORY_ALTERNATIVE);
-            // menu.addIntentOptions(Menu.CATEGORY_ALTERNATIVE, 0, 0, null,
-            // specifics, intent, 0, items);
-
-            // Workaround to add icons:
-            MenuIntentOptionsWithIcons menu2 = new MenuIntentOptionsWithIcons(
-                    this, menu
-            );
-            menu2.addIntentOptions(
-                    Menu.CATEGORY_ALTERNATIVE, 0, 0, null,
-                    specifics, intent, 0, items
-            );
-
-            // Give a shortcut to the edit action.
-            if (items[0] != null) {
-                items[0].setShortcut('1', 'e');
-            }
-        } else {
-            menu.removeGroup(Menu.CATEGORY_ALTERNATIVE);
-        }
+        // Skip any item-specific actions: those go in context menu, not options menu
+        menu.removeGroup(Menu.CATEGORY_ALTERNATIVE);
```

Comment:
Why did you remove the alternative menu items?

Reply:
Apologies, I removed too much here as part of my conversion from ListActivity.
This has now been restored, based on the code in master.
I validated restoration by running  
`git diff master...HEAD  NotePad/src/main/java/org/openintents/notepad/noteslist/NotesList.java`

I wasn't able to find out much about what the alternative items would do, so I don't have a good way to completely test this. I only did a quick test in the emulator to make sure the context menu appears.